pkg/controller/controllercmd/builder.go

```diff
@@ -66,10 +73,38 @@ func (b *ControllerBuilder) WithFileObserver(reactorFunc func(file string, actio
 		}
 		b.fileObserver = observer
 	}
+	reactorFunc := func(filename string, action fileobserver.ActionType) error {
+		defer close(stopCh)
+		message := fmt.Sprintf("Restart triggered because of %s", action.String(filename))
+		b.eventRecorder.Event("OperatorRestarted", message)
+		glog.Warning(message)
+		return nil
+	}
 	b.fileObserver.AddReactor(reactorFunc, files...)
 	return b
 }
 
+// WithEventRecorder will enable event recording for this controller. To make this work, the 'POD_NAME' environment variable has
+// to be present in system with a name of the pod that is currently running the container.
+func (b *ControllerBuilder) WithEventRecorder() *ControllerBuilder {
+	clientConfig, err := b.getClientConfig()
+	if err != nil {
+		panic(fmt.Sprintf("error getting client config: %v", err))
+	}
+	kubeClient := kubernetes.NewForConfigOrDie(clientConfig)
+	namespace, err := b.getNamespace()
+	if err != nil {
+		panic("unable to read the namespace")
+	}
+	controllerRef, err := events.GetControllerReferenceForCurrentPod(kubeClient.CoreV1().Pods(namespace))
+	if err != nil {
+		panic(fmt.Sprintf("unable to obtain replicaset reference for events: %v", err))
+	}
+
+	b.eventRecorder = events.NewRecorder(kubeClient.CoreV1().Events(namespace), b.componentName, controllerRef)
```

Comment:
Why save it like this?  Why not simply build it directly in `Run` and always pass it through?